packages/ag-charts-enterprise/src/features/zoom/zoom.ts

```diff
@@ -249,89 +255,105 @@ export class Zoom extends _ModuleSupport.BaseModuleInstance implements _ModuleSu
 
     private onDragStart(event: _ModuleSupport.InteractionEvent<'drag-start'>) {
         if (!this.enabled || !this.paddedRect) return;
-        this.canDragSelection = this.paddedRect.containsPoint(event.offsetX, event.offsetY);
+
+        // Determine which ZoomDrag behaviour to use.
+        let newDragState = DragState.None;
+
+        if (this.enableAxisDragging && this.hoveredAxis) {
+            newDragState = DragState.Axis;
+        }
+        // Panning & Selection can only happen inside the series rect:
+        else if (this.paddedRect.containsPoint(event.offsetX, event.offsetY)) {
+            const panKeyPressed = this.isPanningKeyPressed(event.sourceEvent as DragEvent);
+            // Allow panning if either selection is disabled or the panning key is pressed.
+            if (this.enablePanning && (!this.enableSelecting || panKeyPressed)) {
+                this.cursorManager.updateCursor(CURSOR_ID, 'grabbing');
+                newDragState = DragState.Pan;
+            }
+            // Do not allow selection only if fully zoomed in or when the pankey is pressed
+            else {
+                const fullyZoomedIn = this.isMinZoom(definedZoomState(this.zoomManager.getZoom()));
+                if (!fullyZoomedIn && !panKeyPressed) {
+                    newDragState = DragState.Select;
+                }
+            }
+        }
+
+        this.dragState = newDragState;
     }
 
     private onDrag(event: _ModuleSupport.InteractionEvent<'drag'>) {
         if (!this.enabled || !this.paddedRect || !this.seriesRect) return;
 
         this.ctx.interactionManager.pushState(_ModuleSupport.InteractionState.ZoomDrag);
 
-        const sourceEvent = event.sourceEvent as DragEvent;
-
-        const isPrimaryMouseButton = sourceEvent.button === 0;
-        if (!isPrimaryMouseButton) return;
-
-        this.isDragging = true;
-        this.tooltipManager.updateTooltip(TOOLTIP_ID);
-
         const zoom = definedZoomState(this.zoomManager.getZoom());
 
-        if (this.enableAxisDragging && this.hoveredAxis) {
-            const { id: axisId, direction } = this.hoveredAxis;
-            const anchor = direction === _ModuleSupport.ChartAxisDirection.X ? this.anchorPointX : this.anchorPointY;
-            const axisZoom = this.zoomManager.getAxisZoom(axisId);
-            const newZoom = this.axisDragger.update(event, direction, anchor, this.seriesRect, zoom, axisZoom);
-            this.updateAxisZoom(axisId, direction, newZoom);
-            return;
-        }
-
-        // Prevent the user from dragging outside the series rect (if not on an axis)
-        if (!this.paddedRect.containsPoint(event.offsetX, event.offsetY)) {
-            return;
+        switch (this.dragState) {
+            case DragState.Axis:
+                if (this.hoveredAxis) {
```

Comment:
nit: rather than indenting the whole block, `if (!this.hoveredAxis) break;`

Reply:
Done.